kubernetes-examples/envelope-encryption/README.md

```diff
@@ -0,0 +1,83 @@
+# Envelope Encryption Example on Kubernetes.
+
+This example will demonstrate the `EnvelopeEncryption` filter which provides an Encryption-at-Rest solution for the
+Apache Kafka(tm) which is transparent to both clients and brokers.
+
+The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and applies encryption to
+the kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
+
+
+## What will the example demonstrate?
+
+It will:
+
+* deploy an Apache Kafka Cluster to Kubernetes using Strimzi.
+* deploy an instance of Kroxylicious to the same Kubernetes cluster  to proxy the Kafka Cluster with configuration for
+  `EnvelopeEncryption`.
+* deploy an instance of HashiCorp Vault to store the keys used to encrypt the kafka records.
+* use an off-cluster kafka producer and kafka consumer send and receive messages, and demonstrate
+  that the records are indeed encrypted on the server.
+
+Running the example will take about 10 minutes.
+
+## Prerequisites
+
+See [prerequistes](../README.md#prerequisites-to-run-the-kubernetes-examples).
+
+## Running the example
+
+1. Clone Kroxylicious Repository
+    ```shell { comment="# We're going to demonstrate the Envelope Encryption feature of Kroxylicious.\nLet's start by cloning the repo." }
+    git clone https://github.com/kroxylicious/kroxylicious.git
+    ```
+1. Change directory to it. 
+    ```shell
+    cd kroxylicious
+    ```
+1. Run the following script.
+    ```shell { comment="# Now let's bring up minikube and install strimzi and vault. The script will also create a kafka cluster and deploy kroxylicious." }
+    ./scripts/run-example.sh kubernetes-examples/envelope-encryption
```

Comment:
Two thoughts on calling run-example.sh

- i'm questioning the value of the illustrative commands produced by postinstall.sh.  I think the content would be better off in the example's README.  I think its presence of the illustrative commands in the asciicema quite confusing.  
- Instead of the invocation of run-example.sh i'm wondering if it would be better to see the individual steps (vault install, strimzi install, invocate kustomize etc) enumerated.  I think this might give the user a better mental model. It would give us a place to mention that the user could substitute a cloud kafka service in place of strimzi or make use of their existing vault.    At the moment a user could believe that the use of strizmi is required or the vault is somehow special.

Reply:
I think it very much depends on the audience/persona being targeted. I think a single command is really powerful as a high level demo/overview. However people looking at it and thinking about the details of how it fits for their environment probably do want the lower level details. 

The other angle I want to keep prioritising is a focus on running Kroxylicious. So I possibly want to see two "commands" run `the-background-stuff.sh` and `run-kroxylicious.sh`

Reply:
So, I got rid of the illustrative commands and re-recorded the demo.